pymongo/results.py

```diff
@@ -81,6 +81,26 @@ def inserted_id(self) -> Any:
         return self.__inserted_id
 
 
+class ClientInsertOneResult(_WriteResult):
```

Comment:
These classes are also very similar to the existing ones, with the removal of `acknowledged`. Could consolidation also be done here by making `acknowledged` optional?